stellarphot/core.py

```diff
@@ -70,31 +111,50 @@ class Camera:
     <Quantity 0.563 arcsec / pix>
 
     """
-    def __init__(self, gain=1.0 * u.electron / u.adu,
-                 read_noise=1.0 * u.electron,
-                 dark_current=0.01 * u.electron / u.second,
-                 pixel_scale=1.0 * u.arcsec / u.pixel):
-        super().__init__()
-
-        # Check that input has units and if not crap out.
-        if isinstance(gain, u.Quantity):
-            self.gain = gain
-        else:
-            raise TypeError("gain must have units.")
-
-        if isinstance(read_noise, u.Quantity):
-            self.read_noise = read_noise
-        else:
-            raise TypeError("read_noise must have units.")
-
-        if isinstance(dark_current, u.Quantity):
-            self.dark_current = dark_current
-        else:
-            raise TypeError("dark_current must have units.")
-        if isinstance(pixel_scale, u.Quantity):
-            self.pixel_scale = pixel_scale
-        else:
-            raise TypeError("pixel_scale must have units.")
+    gain: QuantityType
+    read_noise: QuantityType
+    dark_current: QuantityType
+    pixel_scale: PixelScaleType
+
+    class Config:
+        validate_all = True
+        validate_assignment = True
+        extra = "forbid"
+        json_encoders = {
+            Quantity: lambda v: f"{v.value} {v.unit}",
+            QuantityType: lambda v: f"{v.value} {v.unit}",
+            PixelScaleType: lambda v: f"{v.value} {v.unit}"
+        }
+
+    # When the switch to pydantic v2 happens, this root_validator will need
+    # to be replaced by a model_validator decorator.
+    @root_validator()
+    def validate_gain(cls, values):
```

Comment:
Also, `root_validator` is a classmethod accoring to the documentation, so let's mark it that way:

```suggestion
    @classmethod
    def validate_gain(cls, values):
```

Reply:
[The documentation I looked at](https://docs.pydantic.dev/1.10/usage/validators/) never shows `@classmethod` added in addition to `@root_validator`, can you point mt to your documentation reference?

Reply:
OK, you win, changes made.